tests/docker_scripts/stress_tests.lib

```diff
@@ -263,8 +263,12 @@ function check_logs_for_critical_errors()
     # Remove file logical_errors.txt if it's empty
     [ -s /test_output/logical_errors.txt ] || rm /test_output/logical_errors.txt
 
-    # No such key errors (ignore a.myext which is used in 02724_database_s3.sh and does not exist)
-    rg --text "Code: 499.*The specified key does not exist" /var/log/clickhouse-server/clickhouse-server*.log | grep -v "a.myext" > /test_output/no_such_key_errors.txt \
+    # ignore:
+    #  - a.myext which is used in 02724_database_s3.sh and does not exist
+    #  - "DistributedCacheTCPHandler" and "caller id: None:DistribCache" because they happen inside distributed cache server
+    #  - "ReadBufferFromDistributedCache" exception printed internally by ReadBufferFromDistributedCache, exception will be rethrown and handled correctly
+    #  - "ReadBufferFromS3" exception printed internally by ReadBufferFromS3, exception will be rethrown and handled correctly
+    rg --text "Code: 499.*The specified key does not exist" /var/log/clickhouse-server/clickhouse-server*.log | grep -v -e "a.myext" -e "DistributedCacheTCPHandler" -e "ReadBufferFromDistributedCache" -e "ReadBufferFromS3" -e "caller id: None:DistribCache" > /test_output/no_such_key_errors.txt \
```

Comment:
Also `ReadBufferFromAzureBlobStorage`?

Reply:
Potentially the exception can also be logged here https://github.com/ClickHouse/ClickHouse/blob/12718ff0247365cd06f72e44b5d6b2a550276c4d/src/Disks/IO/AsynchronousBoundedReadBuffer.cpp#L360-L370
So probably change `tryLogCurrentException(__PRETTY_FUNCTION__)` -> `tryLogCurrentException(log)` and add `AsynchronousBoundedReadBuffer` to the list? Though here we indeed do not rethrow for obvious reasons, so may be questionable that we should ignore it or not.